common/display/displayplanemanager.cpp

```diff
@@ -458,6 +460,39 @@ void DisplayPlaneManager::ValidateCachedLayers(
   *render_layers = needs_gpu_composition;
 }
 
+bool DisplayPlaneManager::ValidateSurfaceDamage(
+    const std::vector<OverlayLayer> *layers,
+    const std::vector<OverlayLayer> *previous_layers) {
+  auto layer_begin = layers->begin();
+  auto layer_end = layers->end();
+  bool validate = true;
+  size_t num_layers = layers->size();
+
+  for (size_t i = 0; i <= num_layers; i++) {
+    const HwcRegion &current_surface_damage = layers->at(i).GetSurfaceDamage();
+
+    if (current_surface_damage.kNumRects == 1) {
+      const HwcRect<int> *rect = current_surface_damage.kRects;
+      if ((rect->top == 0) && (rect->bottom == 0) && (rect->left == 0) &&
+          (rect->right == 0)) {
+        const HwcRect<int> &previous = previous_layers->at(i).GetDisplayFrame();
+        const HwcRect<int> &current = layers->at(i).GetDisplayFrame();
+        if ((previous.left != current.left) || (previous.top != current.top)) {
+          validate = true;
+          break;
+        }
+        validate = false;
+        continue;
+      }
+    } else if (current_surface_damage.kNumRects == 0) {
+      validate = true;
```

Comment:
Can we combine these checks here: https://github.com/01org/IA-Hardware-Composer/blob/master/common/display/displayplanemanager.cpp#L405
You can pass bool damaged_region like we do for render_layers, for this we can just set damagaed_region =true even if one layer needs to be re-composited. This  flag along with render layers can be checked in displayqueue. If both flags are not set u can just swap https://github.com/01org/IA-Hardware-Composer/blob/master/common/display/displayqueue.cpp#L314